File: livingrimoire start here/LivinGrimoire java/src/Skills/special_skills/DiGamificationSkillBundle.java
package Skills.special_skills;

import Auxiliary_Modules.AXGamification;
import Auxiliary_Modules.UniqueResponder;
import LivinGrimoire.Skill;

public class DiGamificationSkillBundle extends DiSkillBundle{
    private final AXGamification axGamification = new AXGamification();
    private int gain = 1;
    private int cost = 2;
    public void setGain(int gain) {
        if (gain >0){
            this.gain = gain;}
    }

    public void setCost(int cost) {
        if (cost > 0){this.cost = cost;}
    }
    public void addGrindSkill(Skill skill) {
        axSkillBundle.addSkill(new GamiPlus(skill, axGamification, gain));
        for (int i = 0; i < 10; i++) {
            notes.get("triggers").addResponse("grind " + skill.skillNotes("triggers"));
        }
    }

    public void addCostlySkill(Skill skill) {
        axSkillBundle.addSkill(new GamiMinus(skill, axGamification, cost));
        for (int i = 0; i < 10; i++) {
            notes.get("triggers").addResponse("grind " + skill.skillNotes("triggers"));
        }
    }


    public AXGamification getAxGamification() {
        return axGamification;
    }
    @Override
    public void setDefaultNote() {
        notes.put("notes", new UniqueResponder("a bundle of grind and reward skills"));
    }

}

File: livingrimoire start here/LivinGrimoire Kotlin/kotlin/skills/special/DiGamificationSkillBundle.kt
package skills.special

import auxiliary_modules.AXGamification
import livinGrimoire.DiSkillV2

class DiGamificationSkillBundle : DiSkillBundle() {
    val axGamification = AXGamification()
    private var gain = 1
    private var cost = 2
    fun setGain(gain: Int) {
        if (gain > 0) {
            this.gain = gain
        }
    }

    fun setCost(cost: Int) {
        if (cost > 0) {
            this.cost = cost
        }
    }

    fun addGrindSkill(skill: DiSkillV2) {
        axSkillBundle.addSkill(GamiPlus(skill, axGamification, gain))
    }

    fun addCostlySkill(skill: DiSkillV2) {
        axSkillBundle.addSkill(GamiMinus(skill, axGamification, cost))
    }
}